gtda/plotting/persistence_diagrams.py

```diff
@@ -37,23 +37,26 @@ def plot_diagram(diagram, homology_dimensions=None, plotly_params=None):
         homology_dimensions = np.unique(diagram[:, 2])
 
     diagram_no_dims = diagram[:, :2]
-    max_birth, max_death = np.max(
-        np.where(np.isposinf(diagram_no_dims), -np.inf, diagram_no_dims),
-        axis=0
+    max_val_display = np.max(
+        np.where(np.isposinf(diagram_no_dims), -np.inf, diagram_no_dims)
         )
-    min_birth, min_death = np.max(
-        np.where(np.isneginf(diagram_no_dims), np.inf, diagram_no_dims),
-        axis=0
+    min_val_display = np.min(
+        np.where(np.isneginf(diagram_no_dims), np.inf, diagram_no_dims)
         )
+    parameter_range = max_val_display - min_val_display
+    extra_space = 0.02 * parameter_range
+    min_val_display -= extra_space
+    max_val_display += extra_space
 
     fig = gobj.Figure()
-    fig.add_trace(gobj.Scatter(x=[100 * min(-np.abs(max_death), min_birth),
-                                  100 * max_death],
-                               y=[100 * min(-np.abs(max_death), min_birth),
-                                  100 * max_death],
-                               mode="lines",
-                               line=dict(dash="dash", width=1, color="black"),
-                               showlegend=False, hoverinfo="none"))
+    fig.add_trace(gobj.Scatter(
+        x=[min_val_display, max_val_display],
```

Comment:
This leads to better behaviour when clicking "Autoscale" on the `plotly` HTML object in a notebook. Currently, the factor of 100 means that autoscaling leads to huge ranges..